### PR TITLE
Adding a GNEB implementation in the SPIN package

### PR DESCRIPTION
**Pull request by julient31**
**Summary**

This pull request adds an implementation of the Geodesic Nudged Elastic Band (GNEB) methodology into LAMMPS. 
It adds the neb_spin.cpp and fix_neb_spin.cpp files in the SPIN repository.
Those options were documented.
Examples are provided in examples/SPIN/gneb/.

**Related Issues**

Improvement of the SPIN package.

**Author(s)**

Julien Tranchida, Sandia Nat. Labs, jtranch@sandia.gov

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

- documentation was built and checked with make spelling
- the code was built with the -Wall -Wextra and unused variables were removed
- the code was succesfully compiled and tested with the files provided in examples (iron and skyrmion calculations) and the results were interpolated with the routine provided in examples/SPIN/gneb/interpolate/

**Post Submission Checklist**

_Please check the fields below as they are completed **after** the pull request has been submitted. Delete lines that don't apply_

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] A package specific README file has been included or updated
- [x] One or more example input decks are included

**Further Information, Files, and Links**

_Put any additional information here, attach relevant text or image files, and URLs to external sites (e.g. DOIs or webpages)_

### CODE REVIEW

**Comment by julient31** [comment]
Thanks a lot for the revisions.
- I transferred the interpolation tool toward a new directory called tools/spin/interpolate_gneb. I modified doc/src/neb_spin.txt to account for this change.
- I added the neb/spin command in all the Command*.txt files in doc/src (and also in Howto_replica.txt, Packages_details.txt, fixes.txt).
- I also slightly modified the error messages in neb_spin.cpp to make them more explicit.

File: examples/SPIN/gneb/iron/in.gneb.iron [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+thermo_style    custom step time v_magx v_magz v_magnorm etotal
+thermo_modify   format float %20.15g
+
+compute 	outsp all property/atom spx spy spz sp fmx fmy fmz
+variable	u universe 1 2 3 4
+dump 		1 all custom 200 dump.$u type x y z c_outsp[1] c_outsp[2] c_outsp[3]
+
+min_style	spin
+min_modify 	alpha_damp 1.0 discrete_factor 10.0
+neb/spin	1.0e-12 1.0e-12 200000 100000 1000 final final.iron_spin verbose
```
Comment (akohlmey) [inline examples/SPIN/gneb/iron/in.gneb.iron]: This input fails for me with the following error message:
```
ERROR: Incorrect initial rotation operation (../neb_spin.cpp:633)
Last command: neb/spin	1.0e-12 1.0e-12 200000 100000 1000 final final.iron_spin verbose
```

File: src/SPIN/fix_neb_spin.cpp [hunk 2]
```diff
@@ -0,0 +1,1008 @@
...
+
+
+  // case 1 => to be done
+
+  // single proc per replica
+  // all atoms are NEB atoms and no atom sorting
+  // direct comm of x -> xprev and x -> xnext
+
+  if (cmode == SINGLE_PROC_DIRECT) {
+    if (ireplica > 0)
```
Comment (akohlmey) [inline src/SPIN/fix_neb_spin.cpp]: These if statements look suspicious and the indentation does not match the logic of then code. if only one MPI call should be covered by the ifs, then the second MPI call should be properly indented (i.e. 2 spaces less). if both MPI calls should be covered by the ifs, then there must be curly braces added.

**Comment by akohlmey** [comment]
@julient31 Made some cosmetic changes and removed the unused special copy constructor, which was likely carried over from the neb.cpp code. thus please first pull before adding/committing/pushing new changes.

**Comment by julient31** [comment]
Thanks a lot for the revisions Axel.
- the first one was a silly mistake from me. The loop starting line 31 of neb_spin.cpp is used in case knormsq is equal to zero (which should never be the case). Thus, knormsq needs to be re-evaluated within this loop.
- good catch for the second one. I created the code from fix_neb.cpp, and forgot to change the braces.

**Pull request merged**